HerPublicWebsite/Controllers/QuestionnaireController.cs

```diff
@@ -453,17 +453,57 @@ public async Task<IActionResult> Eligible_Post(EligibleViewModel viewModel)
 
         return RedirectToNextStep(nextStep, viewModel.EntryPoint);
     }
-    
+
     [HttpGet("confirmation")]
     public IActionResult Confirmation_Get()
     {
         return RedirectToAction(nameof(StaticPagesController.Index), "StaticPages");
     }
 
     [HttpGet("ineligible")]
-    public IActionResult Ineligible_Get()
+    public IActionResult Ineligible_Get(bool submitted = false)
     {
-        return RedirectToAction(nameof(StaticPagesController.Index), "StaticPages");
+        var questionnaire = questionnaireService.GetQuestionnaire();
+        var viewModel = new IneligibleViewModel
+        {
+            FoundEpcIsTooHigh = questionnaire.FoundEpcBandIsTooHigh && (questionnaire.EpcDetailsAreCorrect ?? true),
+            IncomeBand = questionnaire.IncomeBand,

```

Comment:
Similarly, the logic for whether the household income is too high should live in the Questionnaire class, not the cshtml file.
So ideally this should be something like `IncomeIsTooHigh`
The property on the Questionnaire class could also take LSOA into account.

Alternatively, the Questionnaire could have a method that returns a set of flags for ineligibility reasons